testlapack/dgehrd.go

```diff
@@ -0,0 +1,230 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+type Dgehrder interface {
+	Dgehrd(n, ilo, ihi int, a []float64, lda int, tau, work []float64, lwork int)
+}
+
+func DgehrdTest(t *testing.T, impl Dgehrder) {
+	rnd := rand.New(rand.NewSource(1))
+
+	// Randomized tests for small matrix sizes that will most likely
+	// use the unblocked algorithm.
+	for _, n := range []int{1, 2, 3, 4, 5, 10, 34} {
+		for _, extra := range []int{0, 1, 13} {
+			for _, optwork := range []bool{true, false} {
+				for cas := 0; cas < 100; cas++ {
+					ilo := rnd.Intn(n)
+					ihi := rnd.Intn(n)
+					if ilo > ihi {
+						ilo, ihi = ihi, ilo
+					}
+					testDgehrd(t, impl, n, ilo, ihi, extra, optwork, rnd)
+				}
+			}
+		}
+	}
+
+	// Selected tests for larger matrix sizes to test the blocked
```

Comment:
I read "Selected tests" as *noun* *verb* rather than *adj* *noun* (I think because of the normal Go documentation style - it seems to have broken my general English parsing). Maybe rephrase as "These are selected tests...".